src/commands/anony.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  Client,
+  CommandInteraction,
+  EmbedBuilder,
+  SlashCommandBuilder,
+  TextChannel,
+} from 'discord.js';
+import ANONY_CONSTANTS from '../constants/anony';
+
+export const anonyCommand = {
+  data: new SlashCommandBuilder()
+    .setName(ANONY_CONSTANTS.name)
+    .setDescription(ANONY_CONSTANTS.description)
+    .addStringOption((option) =>
+      option
+        .setName(ANONY_CONSTANTS.subCommandTitleName)
+        .setDescription(ANONY_CONSTANTS.subCommandTitleDescription)
+        .setRequired(true),
+    )
+    .addStringOption((option) =>
+      option
+        .setName(ANONY_CONSTANTS.subCommandMessageName)
+        .setDescription(ANONY_CONSTANTS.subCommandMessageDescription)
+        .setRequired(true),
+    ),
+  async execute(client: Client, interaction: CommandInteraction) {
+    const channel = client.channels.cache.get(
+      process.env.ANONYMOUS_APPROVAL_CHANNEL_ID,
+    );
+    if (channel?.isTextBased) {
+      const embed = new EmbedBuilder()
+        .setColor(ANONY_CONSTANTS.defaultColorCode)
+        .setTitle(
+          `(#${new Date().getTime()}) ${
+            interaction.options.data.find(
+              (v) => v.name === ANONY_CONSTANTS.subCommandTitleName,
+            )?.value || 'Untitled'
+          }`,
+        )
+        .setAuthor({
+          name: `${interaction.user.username}#${interaction.user.id}`,
+          iconURL: interaction.user.displayAvatarURL(),
+        })
+        .addFields({
+          name: ANONY_CONSTANTS.contentLabel,
+          value:
+            (interaction.options.data.find(
+              (v) => v.name === ANONY_CONSTANTS.subCommandMessageName,
+            )?.value as string) || 'No content',
+        });
+      const message = await (channel as TextChannel).send({ embeds: [embed] });
```

Comment:
You can avoid this type assertion by `discord.js` method `isTextBased`

Reply:
```suggestion
      if (!channel.isTextBased()) {
        throw Error("Please use a text-based channel's id for `ANONYMOUS_APPROVAL_CHANNEL_ID`");
      }
      const message = await (channel as TextChannel).send({ embeds: [embed] });
```

Reply:
Damn, I was wondering why this type guard won't work. Didn't know it's a method